src/content/posts/2021/pitfalls-of-unoptimzied-nuget-packages/index.md

```diff
@@ -0,0 +1,177 @@
+---
+title: "Pitfalls of Unoptimized NuGet Packages"
+description: "Avoid using unoptimized .NET assemblies with UnoptimizedAssemblyDetector."
+date: 2021-05-21T16:20:00-04:00
+tags: [".NET", "Assembly", "NuGet", "Optimization"]
+author: "Bruno Garcia"
+showToc: false
+TocOpen: false
+draft: false
+hidemeta: true
+comments: false
+type: posts
+
+cover:
+    image: "snail-cover.jpg"
+    relative: true
+    alt: "A Snail."
+
+resources:
+- src: 'davidfowl-nuget-package-debug-dark.png'
+- src: 'davidfowl-nuget-package-debug-light.png'
+- src: 'dnx-release-build-pack-dark.png'
+- src: 'dnx-release-build-pack-light.png'
+- src: 'unoptimized-assembly-detected.gif'
+
+---
+
+## tl;dr
+
+You might unknowingly be using an unoptimized dependency which could impact the performance of your app.
+Now, it's easy to get a warning if that happens, at build time,
+using [the NuGet package `UnoptimizedAssemblyDetector`](https://www.nuget.org/packages/UnoptimizedAssemblyDetector/):
+
+```xml
+<PackageReference Include="UnoptimizedAssemblyDetector" Version="0.0.5">
+  <PrivateAssets>all</PrivateAssets>
+</PackageReference>
+```
+
+After you add this NuGet package, a warning will be included in your build if any unoptimized assembly is detected.
+
+[![unoptimized-assembly-detected](unoptimized-assembly-detected.gif)](unoptimized-assembly-detected.gif)
+
+The source code of project is [hosted on GitHub](https://github.com/bruno-garcia/unoptimized-assembly-detector) ⭐.
```

Comment:
This really sounds like it could be an intro instead. In my two-sentence-paragraph structure I'd write it like this (general outline):

You might be using unoptimized dependencies. This can affect performance, but dotnet doesn't provide out of the box solution for detecting this type of issues.

To combat this, I've decided to create a package that augments the build pipeline with a simple analyzer that detects unoptimized assemblies. It's open source and can be found on GitHub and downloaded from NuGet.

In this article I will give a bit more background on what is an optimized or unoptimized assembly in .NET and how much and in which scenarios it may affect performance.